lib/archethic_web/controllers/faucet_controller.ex

```diff
@@ -78,55 +78,52 @@ defmodule ArchethicWeb.FaucetController do
     end
   end
 
-  defp transfer(
-         recipient_address,
-         curve \\ Crypto.default_curve()
-       )
-       when is_bitstring(recipient_address) do
-    {gen_pub, _} = Crypto.derive_keypair(@pool_seed, 0, curve)
+  @curve Crypto.default_curve()
+  @spec prepare_transaction(binary()) :: {:ok, Transaction.t()} | {:error, any()}
+  defp prepare_transaction(recipient_address) do
+    pool_gen_address =
+      @pool_seed
+      |> Crypto.derive_keypair(0, @curve)
+      |> elem(0)
+      |> Crypto.derive_address()
 
-    pool_gen_address = Crypto.derive_address(gen_pub)
-
-    with {:ok, last_address} <-
-           Archethic.get_last_transaction_address(pool_gen_address),
+    with {:ok, last_address} <- Archethic.get_last_transaction_address(pool_gen_address),
          {:ok, last_index} <- Archethic.get_transaction_chain_length(last_address) do
-      create_transaction(last_index, curve, recipient_address)
+      {:ok, Transaction.new(:transfer, get_txn_data(recipient_address), @pool_seed, last_index)}
     else
-      {:error, _} = e ->
+      e ->
         e
     end
   end
 
-  defp create_transaction(transaction_index, curve, recipient_address) do
-    tx =
-      Transaction.new(
-        :transfer,
-        %TransactionData{
-          ledger: %Ledger{
-            uco: %UCOLedger{
-              transfers: [
-                %UCOLedger.Transfer{
-                  to: recipient_address,
-                  amount: 10_000_000_000
-                }
-              ]
+  @unit_uco 100_000_000
+  @max_uco 30
+  @uco_limit @max_uco * @unit_uco
+  defp get_txn_data(recipient_address) do
+    %TransactionData{
+      ledger: %Ledger{
+        uco: %UCOLedger{
+          transfers: [
+            %UCOLedger.Transfer{
+              to: recipient_address,
+              amount: @uco_limit
             }
-          }
-        },
-        @pool_seed,
-        transaction_index,
-        curve
-      )
+          ]
+        }
+      }
+    }
+  end
 
-    tx_address = tx.address
-    TransactionSubscriber.register(tx_address, System.monotonic_time())
+  defp send_txn(txn, recipient_address) do
+    txn_addr = txn.address
```

Comment:
You should use pattern matching in the function argument. This will be more elegant ;)

Reply:
👍